containerd-shim-v2/service.go

```diff
@@ -728,12 +728,20 @@ func (s *service) Wait(ctx context.Context, r *taskAPI.WaitRequest) (*taskAPI.Wa
 	//wait for container
 	if r.ExecID == "" {
 		ret = <-c.exitCh
+
+		// refill the exitCh with the container process's exit code in case
+		// there were other waits on this process.
+		c.exitCh <- ret
```

Comment:
Is it the appropriate solution here? What happens in the case the caller invoke `Wait()` from two different threads, but the process is still running? It feels to me this is not gonna go well.
Maybe we should add some larger locking on this function if we want to avoid any potential race condition here.

Reply:
Hi @sboeuf, If more than one Wait() are invoked when the process is still running, all of them will hang on this exitCh channel, until the process exits and then they will be serialized to get the return code.

Reply:
you're right!